gapis/resolve/dependencygraph/footprint.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencygraph
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/core/app/benchmark"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/capture"
+	"github.com/google/gapid/gapis/database"
+)
+
+var footprintBuildCounter = benchmark.GlobalCounters.Duration("footprint.build")
+
+// Footprint contains a list of command and a list of Behaviours which
+// describes the side effect of executing the commands in that list.
+type Footprint struct {
+	Commands                                []api.Cmd
+	Behaviours                              []*Behaviour
+	commandIndexToBehaviourIndexLookupTable *api.SubCmdIdxTrie
```

Comment:
This isn't java. `cmdIdxToBehavior` will suffice.
Also it probably doesn't need to be a pointer type.
Please put private members after public ones.